sdk/synapse/synapse-spark/src/SparkClient.ts

```diff
@@ -0,0 +1,481 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+/* eslint @typescript-eslint/member-ordering: 0 */
+/// <reference lib="esnext.asynciterable" />
+
+import {
+  TokenCredential,
+  PipelineOptions,
+  createPipelineFromOptions,
+  bearerTokenAuthenticationPolicy,
+  operationOptionsToRequestOptionsBase
+} from "@azure/core-http";
+
+import { SynapseSpark } from "./generated";
+import { logger } from "./logger";
+import { SDK_VERSION, DEFAULT_SYNAPSE_SCOPE } from "./constants";
+import { createSpan, getCanonicalCode } from "./tracing";
+
+import {
+  SparkClientOptions,
+  GetSparkBatchJobOptions,
+  ListSparkBatchJobsOptions,
+  CreateSparkBatchJobOptions,
+  CancelSparkBatchJobOptions,
+  GetSparkSessionOptions,
+  ListSparkSessionsOptions,
+  CreateSparkSessionOptions,
+  CancelSparkSessionOptions,
+  ResetSparkSessionTimeoutOptions,
+  GetSparkStatementOptions,
+  ListSparkStatementsOptions,
+  CreateSparkStatementOptions,
+  CancelSparkStatementOptions,
+  SparkBatchJobOptions,
+  SparkSessionOptions,
+  SparkStatementOptions,
+  OperationResponse
+} from "./models";
+
+import {
+  GetSparkBatchJobResponse,
+  ListSparkBatchJobsResponse,
+  CreateSparkBatchJobResponse,
+  GetSparkSessionResponse,
+  ListSparkSessionsResponse,
+  CreateSparkSessionResponse,
+  GetSparkStatementResponse,
+  ListSparkStatementsResponse,
+  CreateSparkStatementResponse
+} from "./models";
+
+export { PipelineOptions, logger };
+
+export class SparkClient {
+  /**
+   * The base URL to the workspace
+   */
+  public readonly workspaceEndpoint: string;
+
+  /**
+   * @internal
+   * @ignore
+   * A reference to the auto-generated synapse spark HTTP client.
+   */
+  private readonly client: SynapseSpark;
+
+  constructor(
```

Comment:
we usually have constructor docs with a code example of how to instantiate the client

Reply:
Got it. Will add it.

---
In reply to: [521613645](https://github.com/Azure/azure-sdk-for-js/pull/12487#discussion_r521613645) [](ancestors = 521613645)